src/Type/Php/MethodExistsTypeSpecifyingExtension.php

```diff
@@ -44,6 +45,11 @@ public function specifyTypes(
 		TypeSpecifierContext $context
 	): SpecifiedTypes
 	{
+		$objectType = $scope->getType($node->args[0]->value);
+		if ((!$objectType instanceof ObjectType) && (!$objectType instanceof ConstantStringType)) {
```

Comment:
Try `(new StringType())->isSuperTypeOf($objectType)->yes()` instead. This will also account for unions of constant strings and is generally better.

Reply:
If I do this then a call like this `method_exists(MethodExists::class, 'undefinedMethod')` will no longer produce the error

> Call to function method_exists() with 'CheckTypeFunctionCa…' and 'undefinedMethod' will always evaluate to false.

Reply:
Yeah, that's correct. Please write another test for that. See https://github.com/phpstan/phpstan/issues/1267 it's also reported.

Reply:
I'm not sure if I got you right. I changed the type check according to your request and removed the expected errors for this call:

`method_exists(MethodExists::class, 'undefinedMethod')`